server/datastore/mysql/migrations/tables/20220711104651_RemoveCpeIDColumnFromSoftwareCve.go

```diff
@@ -0,0 +1,93 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220711104651, Down_20220711104651)
+}
+
+func Up_20220711104651(tx *sql.Tx) error {
+	fmt.Println("Deleting dummy software_cpe entries...")
+
+	var min int
+	var max int
+
+	const selectStmt = `
```

Comment:
This is an interesting way of finding the software cpes that need to be deleted. Is there an advantage over selecting just the IDs and doing `DELETE FROM ... WHERE id IN (?)`? I think this method will result in having to execute many more delete queries than are necessary.

Reply:
One of the requirements for this was to assume that there would be potentially millions of entries in the `software_cve` table - so some kind of batching strategy was needed.

Reply:
You don't really need to select them first either. You can just do a `DELETE ... LIMIT 500` in a loop until `RowsAffected()` returns `0`.


```
	// Remove in batches
	const deleteStmt = `
DELETE FROM software_cpe
WHERE cpe LIKE 'none:%' LIMIT 500`

	for {

		res, err := tx.Exec(deleteStmt)
		if err != nil {
			return errors.Wrapf(err, "deleting dummy software_cpe entries")
		}

		affected, err := res.RowsAffected()
		if err != nil {
			return err
		}

		if affected == 0 {
			break
		}
	}

```